### PR TITLE
fix(aria-valid-attr-value): report empty values as incomplete

### PR DESCRIPTION
**Pull request by WilcoFiers**
There is no real difference between an empty ARIA attribute, a null attribute, and not having the attribute at all. Failing empty ARIA attributes therefor seems a little strict. They shouldn't be passed either though, since it is pretty easy to mistakenly assume `<input aria-invalid>` means the input is invalid. Reporting `incomplete` seems more appropriate. This PR makes that change, along with:

- String type attributes, like aria-valuenow are still not allowed empty
- `aria-invalid` is now reported as incomplete when empty, rather than passed
- Extra tests were added to aria-required-attr to ensure empty attributes when required are failed there

Closes issue: #3489

### CODE REVIEW

File: lib/checks/aria/aria-valid-attr-value-evaluate.js [hunk 1]
```diff
@@ -106,30 +107,35 @@ function ariaValidAttrValueEvaluate(node, options, virtualNode) {
     } catch (e) {
       needsReview = `${attrName}="${attrValue}"`;
       messageKey = 'idrefs';
+      return;
```
Comment (WilcoFiers) [inline lib/checks/aria/aria-valid-attr-value-evaluate.js:110]: Avoid getting into that if statement below, as we don't want both `needsReview` and `invalid` set.

File: lib/checks/aria/aria-valid-attr-value-evaluate.js [hunk 2]
```diff
@@ -106,30 +107,35 @@ function ariaValidAttrValueEvaluate(node, options, virtualNode) {
...
+      if (attrValue === '' && !isStringType(attrName)) {
+        needsReview = attrName;
+        messageKey = 'empty';
+      } else {
+        invalid.push(`${attrName}="${attrValue}"`);
+      }
     }
   });
 
-  if (needsReview) {
```
Comment (WilcoFiers) [inline lib/checks/aria/aria-valid-attr-value-evaluate.js:119]: Report invalid before needsReview.

File: test/checks/aria/aria-required-attr.js [hunk 3]
```diff
@@ -1,48 +1,72 @@
-describe('aria-required-attr', function() {
+describe('aria-required-attr', function () {
```
Comment (WilcoFiers) [inline test/checks/aria/aria-required-attr.js:1]: Check didn't change, just want to better prove empty values are not allowed under this check.

**Review by straker — APPROVED** [review]
